.github/workflows/publish-npm.yml

```diff
@@ -0,0 +1,37 @@
+name: Publish release to npm
+
+on:
+  pull_request:
+    types: 
+      - closed
+    branches: 
+      - main
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    if: |
+      startsWith(github.head_ref, 'release-'')
+      && github.event.pull_request.merged
```

Comment:
So AFAIK there's no concise "on:" syntax for "on merge of a pr to main from a branch like feature-*", but the above should be what we need. 

'startsWith` is an [expression](https://docs.github.com/en/actions/learn-github-actions/expressions) available in Actions.

'github.head_ref' is a property available in the [github context](https://docs.github.com/en/actions/learn-github-actions/contexts)

This ends up looking somewhat complex, but that's effectively what's going on here.